src/python/pants/backend/jvm/tasks/jvm_platform_analysis.py

```diff
@@ -181,8 +185,45 @@ def validate_platform_dependencies(self):
     """
     conflicts = []
 
+    jvm_platform_map = {}
+
+    class JvmPlatformAssignment(datatype([
+        ('from_upstream', bool),
+        ('source', SubclassesOf(JvmTarget)),
+        'version',
+    ])):
+
+      @property
+      def transitive_assignment_description(self):
+        if self.from_upstream:
+          # Version was assigned from a target which depends on the current one.
+          return '{} from <- {} (upstream)'.format(self.version, self.source.address.spec)
+        else:
+          # Version was assigned from a target which the current one depends on.
+          return '{} from -> {} (downstream)'.format(self.version, self.source.address.spec)
+
+    def maybe_get_assigned_platform_version(target):
+      cur_version = self.jvm_version(target)
+      if not cur_version:
+        mapped_assignment = jvm_platform_map.get(target, None)
+        if mapped_assignment:
+          return mapped_assignment.version
+      return cur_version
+
     def is_conflicting(target, dependency):
-      return self.jvm_version(dependency) > self.jvm_version(target)
+      this_version = maybe_get_assigned_platform_version(target)
+      dep_version = maybe_get_assigned_platform_version(dependency)
+
+      if this_version and dep_version:
+        return dep_version > this_version
+      elif this_version and not dep_version:
+        jvm_platform_map[dependency] = JvmPlatformAssignment(True, target, this_version)
+        return False
+      elif dep_version and not this_version:
+        jvm_platform_map[target] = JvmPlatformAssignment(False, dependency, dep_version)
```

Comment:
Could you please use keyword args here? I had to scroll back up to figure out what `False` and `True` meant, which isn't that bad because the definition is close by but also slows down the reader.

Reply:
Yes! This was an unnecessary compression.